src/@@notfound.js

```diff
@@ -1 +1 @@
-__non_webpack_require__('UNKNOWN');
+module.exports = __non_webpack_require__('UNKNOWN');
```

Comment:
What's the purpose of this change?

Reply:
There are cases that the packages are missing at ncc build time, yet presented at global scope or runtime. Without this fix, the `eval("require")('dep')` can succeed at runtime but the result is not exported.

Reply:
> There are cases that the packages are missing at ncc build time, yet presented at global scope or runtime. 

Which cases? Can you add a test that fails but succeeds with this PR?

Reply:
Let's say we have a project referencing an optional dependency using

```js
try {
  foo = require('foo');
} catch {}
```

Package `foo` is not installed on build time since it's totally optional and should not be bundled into the products. So package `foo` will be transformed by loader `runtime-notfound` to `eval("require")("foo")` without assigning return value to `module.exports`. Since then the `require('foo')` can succeed without any error at runtime but the return value is `undefined`, not the dynamically required package module.

The test case `test/unit/runtime-notfound/output.js` is already presented in the repo, and is identical to the case above.

Reply:
Sorry I should have been more clear. Can you add an **integration** test that fails with the latest ncc but succeeds with this PR?

Reply:
Integration test added, PTAL :)